### ISSUE TITLE
Problem getting opmrun to run

### ISSUE THREAD

**Issue by joakim-hove** [OPM/opm-utilities#57]
```
bash% python omprun.py
OPMRUN Cannot Import PySimpleGUI - Please Install Using pip3
```
```
bash%pip3 install PySimpleGUIpip3 install PySimpleGUI
Collecting PySimpleGUI
  Downloading https://files.pythonhosted.org/packages/b1/43/ea5d29f303241c6308b0dc7d684aab90763a7e01d011f1ec819da9ca94d0/PySimpleGUI-4.19.0-py3-none-any.whl (333kB)
     |████████████████████████████████| 337kB 4.7MB/s 
Installing collected packages: PySimpleGUI
Successfully installed PySimpleGUI-4.19.0
```
```
bash% python omprun.py
OPMRUN Cannot Import PySimpleGUI - Please Install Using pip3
```

**Comment by OPMUSER** [comment]
Will take a look at this now, what distro are you using and what version 
of Python?

OPM User
opmuser0@gmail.com

**Comment by joakim-hove** [comment]
> Will take a look at this now, what distro are you using and what version
of Python?

I have Python 3.7.3 on Debian 10.4

**Comment by OPMUSER** [comment]
Joakim,

Okay the issue is that the install went to a location that was not 
picked up by the check. So for a workaround uninstall the package and 
then re-install with:

|"python3 -m pip install --user ||*package*||'"|

I have re-factored the code (not just for this) and will issue a pull 
request after I have tested the code a bit more. The code change for 
this issue is now:
|#
# Import Required Non-Standard Modules
#
starterr = False
required = {'airspeed', 'pandas', 'pyDOE2', 'PySimpleGUI'}
for package in required:
     try:
         dist = pkg_resources.get_distribution(package)
         if package == 'PySimpleGUI':
             sg = importlib.import_module(package)
         if package == 'pandas':
             pd = importlib.import_module(package)
         else:
             importlib.import_module(package)
         print('Startup: Require Module - ' + dist.key + '(' + 
dist.version + ') Imported')
     except pkg_resources.DistributionNotFound:
         print('Startup: Import Require Package - ' + package + ' Failed')
         print('Startup: Use "python3 -m pip install --user ' + package 
+ '" to install')
         starterr = True

if starterr:
     print('Startup: Use "python3 -m pip -- verbose list" to get a list 
of installed packages')
     exit('Startup: Failed due to Missing Module(s)')|

and this appears to work on a clean install of Ubuntu, but I want to 
test on Debian as well just to make sure.

Yes I know, you want a pip install for the package, it is on the to do 
list.

OPM User
opmuser0@gmail.com

**Comment by joakim-hove** [comment]
> Okay the issue is that the install went to a location that was not
picked up by the check. So for a workaround uninstall the package and
then re-install with: `python3 -m pip install --user ||*package*||'`

This sound very weird to me; if you use `pip install package` and then just `import package` it should work - no need to specify `--user` install - that is a symptom of something else which is wrong - sorry.

I am sorry - but this:

```Python
for package in required:
     try:
         dist = pkg_resources.get_distribution(package)
         if package == 'PySimpleGUI':
             sg = importlib.import_module(package)
         if package == 'pandas':
             pd = importlib.import_module(package)
```
is the wrong approach. The `importlib` package is a low level package implementing the `import` statement; it is exposed in standard to facilitate specialized import functionality - *not* for run of the mill imports like you have here.

I don't know the details - but I am skeptical to the `PySimpleGUI` package, that seems to be a frontend to at least three different gui backends - namely `Tk`, `Qt` and `Wx` - to me that seems to be a fragile high maintenance solution. I do not have `Qt` or `Wx` installed - i.e. I guess `PySimpleGUI` falls back to `Tk` in my case - which gui backend do you use?

**Comment by discover-volve** [comment]
I face the same issue in running OPMRUN in Ubuntu 20.04 . I'm using the default installation of Python in Ubuntu.

**Comment by OPMUSER** [comment]
Well it does work for Ubuntu 20.04 in my VM, so without further information it is difficult to see why it is not working for you.  

Have you installed tkinter? On Linux based systems install tkinter via the package manager.  Debian versions of Linux you have to install it manually by using the following command:` sudo apt-get install python3-tk`

The PR checks for this and issues the above message, but the PR has not been accepted in the master. I'll send you my version of OPMRUN as a zip file to your email address.

Let me know how you get one.

**Comment by discover-volve** [comment]
Hi, 

I made it work in my Virtual-box machine with Ubuntu 20.04. Here are the
issues,  

The PySimpleGUI library need to be installed with: sudo apt-get
python3.8-tk. Also, airspeed needs to be separately installed.  

I need to run OPMRUN via the terminal: python3 opmrun.py  

The entire application closes if I close the terminal. Also, the
application closes when I click the 'x' button for any of the open
windows. For example, the "Add data file" window.  

I had to comment out the code for the error box message that pops up
when the python version is above 3.7.4. The default python version is
3.8.4 in Ubuntu 20.04 . 

I'm tying to make a OPM FLOW tutorial for the Volve Oilfield for my
blog, DiscoverVolve.com and understanding OPMRUN makes it very
user-friendly. 

Congratulations on your wonderful effort towards contributing to the
open-source community. I'm looking forward towards receiving your zip
file. 

Regards, 

Ankit Bansal

On 2020-07-01 08:24, OPMUSER wrote:

> Well it does work for Ubuntu 20.04 in my VM, so without further information it is difficult to see why it is not working for you. 
> 
> Have you installed tkinter? On Linux based systems install tkinter via the package manager. Debian versions of Linux you have to install it manually by using the following command: sudo apt-get install python3-tk 
> 
> The PR checks for this and issues the above message, but the PR has not been accepted in the master. I'll send you my version of OPMRUN as a zip file to your email address. 
> 
> Let me know how you get one. 
> 
> --
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub [1], or unsubscribe [2].
 

Links:
------
[1]
https://github.com/OPM/opm-utilities/issues/57#issuecomment-652156916
[2]
